lib/generators/accessly/install/templates/db/migrate/create_permitted_action_on_objects.rb

```diff
@@ -0,0 +1,14 @@
+class CreatePermittedActionOnObjects < ActiveRecord::Migration<%= migration_version %>
+  def change
+    create_table :accessly_permitted_action_on_objects, force: true, id: :uuid do |t|
+      t.column :segment_id, :integer, default: -1
+      t.column :action, :integer, null: false
+      t.column :actor_id, :integer, null: false
+      t.column :actor_type, :string, null: false
+      t.column :object_type, :string, null: false
+      t.column :object_id, :integer, null: false
+    end
+
+    add_index(:accessly_permitted_action_on_objects, [:segment_id, :action, :actor_id, :actor_type, :object_type, :object_id], unique: true, name: "acessly_paoo_uniq_table_idx")
```

Comment:
### Get a list of lessons a user has permissions on and the permissions they have on each

#### Why
On the `/contents` page, we'll want to show all the lessons the user has permission to see and perhaps show some additional controls based on their permissions (like an archive button, edit button, etc).

#### SQL
```sql
SELECT object_id, object_type, action
FROM accessly_permitted_action_on_objects
WHERE segment_id = $1 AND actor_type = $2 AND actor_id = $3 AND object_type IN ('Lesson', 'LearningPaths::Path')
```

#### Index that might help
```rb
[:segment_id, :actor_type, :actor_id, :object_type]
```

Actually, the index I proposed previously would also work here:

```rb
[:segment_id, :actor_type, :actor_id, :object_type, :object_id]
```